.github/workflows/test-package-build.yml

```diff
@@ -0,0 +1,94 @@
+name: Test package building
+
+on:
+  push:
+    branches:
+      - master
+      - maint/*
+    tags:
+      - "*"
+  pull_request:
+    branches:
+      - master
+      - maint/*
+defaults:
+  run:
+    shell: bash
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - uses: actions/setup-python@v4
+        with:
+          python-version: 3.9
+      - run: pip install --upgrade build twine
+      - name: Build sdist and wheel
+        run: python -m build
+      - run: twine check dist/*
+      - name: Upload sdist and wheel artifacts
+        uses: actions/upload-artifact@v3
+        with:
+          name: dist
+          path: dist/
+      - name: Build git archive
+        run: mkdir archive && git archive -v -o archive/archive.tgz HEAD
+      - name: Upload git archive artifact
+        uses: actions/upload-artifact@v3
+        with:
+          name: archive
+          path: archive/
+  test-package:
+    runs-on: ubuntu-latest
+    needs: [build]
+    strategy:
+      matrix:
+        package: ['wheel', 'sdist', 'archive']
+    steps:
+      - name: Download sdist and wheel artifacts
+        if: matrix.package != 'archive'
+        uses: actions/download-artifact@v3
+        with:
+          name: dist
+          path: dist/
+      - name: Download git archive artifact
+        if: matrix.package == 'archive'
+        uses: actions/download-artifact@v3
+        with:
+          name: archive
+          path: archive/
+      - uses: actions/setup-python@v4
+        with:
+          python-version: 3.9
+      - name: Display Python version
+        run: python -c "import sys; print(sys.version)"
+      - name: Update pip
+        run: pip install --upgrade pip
+      - name: Install wheel
+        if: matrix.package == 'wheel'
+        run: pip install dist/*.whl
+      - name: Install sdist
+        if: matrix.package == 'sdist'
+        run: pip install dist/*.tar.gz
+      - name: Install archive
+        if: matrix.package == 'archive'
+        run: pip install archive/archive.tgz
+      - name: Install test extras
+        run: pip install spyglass[test]
```

Comment:
```suggestion
        run: pip install .[test]
```

We could install the test extras from PyPI using `pip install spyglass-neuro[test]` but that might not contain any changes to the test extras from the current PR (and won't work in this PR because the test extras don't exist). To use the latest test extras of the current PR, we could do `pip install .[test]`